pytorch_lightning/trainer/connectors/logger_connector/result.py

```diff
@@ -370,10 +370,10 @@ def extra(self, extra: Mapping[str, Any]) -> None:
 
         def check_fn(v):
             if v.grad_fn is not None:
-                warning_cache.warn(
+                warning_cache.deprecation(
```

Comment:
name suggestion "deprecate" in line with "warn"

Reply:
We already have `rank_zero_deprecation` instead of `rank_zero_deprecate`. Just following that

Reply:
so warn -> warning then?

Reply:
I don't mind which one we go with, but needs to be done in a separate PR as `warning_cache.deprecation` was added in 1/2